### PR TITLE
[CL-389] Exclude end slot label content from truncation

### PR DESCRIPTION
**Pull request by vleague2**
## 🎟️ Tracking

[CL-389](https://bitwarden.atlassian.net/browse/CL-389)

## 📔 Objective

This PR updates the `label` directive to be a component instead. This allows us to better position the content inside the label by adding an `end` slot. Content in the `end` slot will not be truncated if the label is longer than its container, while the default content will be truncated. This change is needed because form field labels are now placed inside the form field border, which constrains the available space and makes truncation more likely on smaller screens/containers. Checkboxes and radio buttons will continue to wrap, although any end slot content should still be specified with `slot="end"`.

This PR also updates the order of the HTML elements in form fields to provide a more intuitive tab order. Previously, prefix and suffix buttons were focusable before any label content. Now, label content will be focusable first.

## 📸 Screenshots

See Storybook Publish action for more. (Field and Checkbox both have long label stories.)

Screenshot 2024-08-14 at 1 48 37 PM

Screenshot 2024-08-12 at 4 29 03 PM

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

[CL-389]: https://bitwarden.atlassian.net/browse/CL-389?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/components/src/form-control/label.component.html [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<!-- labels inside a form control (checkbox, radio button) should not truncate -->
+<span class="[&:not(bit-form-control_*)]:tw-truncate">
```
Comment (vleague2) [inline libs/components/src/form-control/label.component.html:2]: I originally was using just this one line with css for things related to being inside a form control. Now that I need the actual function for the `ngIf` renders, is it weird to have the css too?

Comment (willmartian) [reply libs/components/src/form-control/label.component.html:2]: Yeah, I think using one technique is easier to understand vs mixing both.

File: libs/components/src/form-control/label.component.html [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    <ng-container *ngTemplateOutlet="endSlotContent"></ng-container>
+  </ng-container>
+</span>
+<ng-container *ngIf="!isInsideFormControl">
+  <ng-container *ngTemplateOutlet="endSlotContent"></ng-container>
+</ng-container>
+
+<ng-template #endSlotContent>
+  <ng-content select="[slot=end]"></ng-content>
+</ng-template>
```
Comment (willmartian) [inline libs/components/src/form-control/label.component.html:14]: Minor nit: Could you move this template to top of file? That way, whenever we see `*ngTemplateOutlet="endSlotContent"`, we already have the context of what it is referring to.

File: libs/components/src/form-field/form-field.component.html [hunk 3]
```diff
@@ -16,6 +16,30 @@
 </ng-template>
 
 <div *ngIf="!readOnly; else readOnlyView" class="tw-w-full tw-relative tw-group/bit-form-field">
+  <div class="tw-absolute tw-w-full tw-h-full tw-top-0 tw-pointer-events-none tw-z-20">
```
Comment (willmartian) [inline libs/components/src/form-field/form-field.component.html:19]: We move this code block and just add the z index? Any other changes?

Comment (vleague2) [reply libs/components/src/form-field/form-field.component.html:19]: Surprisingly no! I guess all the work we did to simplify the code made a change like this pretty trivial lol

**Review by willmartian — APPROVED** [review]

**Pull request merged**